[32mINFO    [0m juju.model:model.py:2098 Deploying ch:amd64/jammy/self-signed-certificates-155
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: Primary
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] active: 
  config-server/1 [idle] active: Primary
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/1 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/2 [idle] blocked: Shard requires TLS to be enabled.
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: Primary
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/1 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/2 [idle] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] maintenance: enabling TLS
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] maintenance: enabling TLS
  shard-one/1 [idle] active: 
  shard-one/2 [executing] active: 
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [idle] active: Primary
  shard-one/2 [idle] active: 
  shard-two/0 [idle] active: Primary
  config-server/0 [executing] active: 
  config-server/1 [executing] active: Primary
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/1 [idle] active: Primary
  shard-one/2 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] active: 
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/1 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/2 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-two/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  config-server/0 [executing] active: Primary
  config-server/1 [executing] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/1 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/2 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-two/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  config-server/0 [executing] active: 
  config-server/1 [executing] maintenance: enabling TLS
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/1 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:834 Model status:

Model  Controller          Cloud/Region        Version  SLA          Timestamp
test   microk8s-localhost  microk8s/localhost  3.4.4    unsupported  02:12:15Z

App                                Version  Status   Scale  Charm                     Channel        Rev  Address         Exposed  Message
config-server                               active       2  mongodb-k8s                                3  10.152.183.59   no       Primary
self-signed-certificates                    active       1  self-signed-certificates  latest/stable  155  10.152.183.111  no       
self-signed-certificates-separate           active       1  self-signed-certificates  latest/stable  155  10.152.183.229  no       
shard-one                                   waiting      3  mongodb-k8s                                4  10.152.183.142  no       installing agent
shard-two                                   waiting      1  mongodb-k8s                                5  10.152.183.151  no       installing agent

Unit                                  Workload  Agent  Address       Ports  Message
config-server/0*                      active    idle   10.1.225.161         Primary
config-server/1                       active    idle   10.1.225.162         
self-signed-certificates-separate/0*  active    idle   10.1.225.175         
self-signed-certificates/0*           active    idle   10.1.225.154         
shard-one/0*                          blocked   idle   10.1.225.169         Shard CA and Config-Server CA don't match.
shard-one/1                           blocked   idle   10.1.225.173         Shard CA and Config-Server CA don't match.
shard-one/2                           blocked   idle   10.1.225.170         Shard CA and Config-Server CA don't match.
shard-two/0*                          blocked   idle   10.1.225.174         Shard CA and Config-Server CA don't match.

[32mINFO    [0m pytest_operator.plugin:plugin.py:840 Juju error logs:

unit-config-server-0: 01:34:33 ERROR unit.config-server/0.juju-log database-peers:0: Failed to create the operator user: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 926, in _init_operator_user
    process.wait_output()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 1771, in wait_output
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)
ops.pebble.ExecError: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
unit-config-server-0: 01:34:39 ERROR unit.config-server/0.juju-log database-peers:0: Failed to create the operator user: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 926, in _init_operator_user
    process.wait_output()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 1771, in wait_output
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)
ops.pebble.ExecError: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
unit-config-server-0: 01:36:07 ERROR unit.config-server/0.juju-log config-server:3: Failed to add shard shard-one to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1724290567, 1), 'signature': {'hash': b'\\x8f\\xae\\xe3\\xdbq)\\x9c,\\xf9D\\x9cf\\n\\xb8\\xecS\\xdbi\\xe4\\xb0', 'keyId': 7405771237584011288}}, 'operationTime': Timestamp(1724290567, 1)}")
unit-config-server-0: 01:36:07 ERROR unit.config-server/0.juju-log config-server:3: shard-one shard does not have the same auth as the config server.
unit-config-server-0: 01:36:24 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724290583, 1), \'signature\': {\'hash\': b\'\\xfd\\xc2Q&\\xb2F\\x86qo\\xc14\\xe9\\x07\\xb2b\\\\\\xdf\\xd1\\x02\\x15\', \'keyId\': 7405771237584011288}}, \'operationTime\': Timestamp(1724290583, 1)}')
unit-config-server-0: 01:36:24 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard-one to cluster
unit-config-server-0: 01:36:24 ERROR unit.config-server/0.juju-log config-server:4: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724290583, 1), \'signature\': {\'hash\': b\'\\xfd\\xc2Q&\\xb2F\\x86qo\\xc14\\xe9\\x07\\xb2b\\\\\\xdf\\xd1\\x02\\x15\', \'keyId\': 7405771237584011288}}, \'operationTime\': Timestamp(1724290583, 1)}')
unit-config-server-0: 01:36:35 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard shard-two to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1724290594, 3), 'signature': {'hash': b'?\\xe6\\xbe\\xd0\\xb3\\x83R\\x90\\x9e\\x1a&\\xf8\\x04<\\xecZ\\x10\\xf7\\xc5\\xa3', 'keyId': 7405771237584011288}}, 'operationTime': Timestamp(1724290594, 3)}")
unit-config-server-0: 01:36:35 ERROR unit.config-server/0.juju-log config-server:4: shard-two shard does not have the same auth as the config server.
unit-shard-one-1: 01:40:40 ERROR unit.shard-one/1.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 01:40:40 ERROR unit.shard-one/0.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-two-0: 01:40:40 ERROR unit.shard-two/0.juju-log certificates:6: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 01:40:40 ERROR unit.config-server/1.juju-log certificates:7: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 01:40:41 ERROR unit.shard-one/2.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 01:40:41 ERROR unit.shard-one/1.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:40:41 ERROR unit.config-server/0.juju-log certificates:7: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-two-0: 01:40:41 ERROR unit.shard-two/0.juju-log certificates:6: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-0: 01:40:41 ERROR unit.shard-one/0.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-2: 01:40:41 ERROR unit.shard-one/2.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:40:41 ERROR unit.config-server/1.juju-log certificates:7: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:40:42 ERROR unit.config-server/0.juju-log certificates:7: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:40:56 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:40:56 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:41:11 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:41:11 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-shard-one-0: 01:41:53 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c69760c8d4b0bcb5ebd12e, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0005546149999418049>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.00048096499995153863>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0005065230000127485>]>
unit-shard-one-0: 01:41:56 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c697634292fb1ed6b6bf5c, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0005482530000335828>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=NotPrimaryError("The server is in quiesce mode and will shut down, full error: {'topologyVersion': {'processId': ObjectId('66c6975cc22c399da9430ac6'), 'counter': 4}, 'ok': 0.0, 'errmsg': 'The server is in quiesce mode and will shut down', 'code': 91, 'codeName': 'ShutdownInProgress', 'remainingQuiesceTimeMillis': 13943, 'lastCommittedOpTime': Timestamp(0, 0), '$clusterTime': {'clusterTime': Timestamp(1724290912, 2), 'signature': {'hash': b'jr\\x01-\\x1b\\x03\\xec\\xc3\\xcd\\xbb\\x9b\\xdf\\xea>\\xfbU\\xc7h-\\xa7', 'keyId': 7405771237584011288}}, 'operationTime': Timestamp(1724290845, 1)}")>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0005211330000065573>]>
unit-shard-one-0: 01:41:58 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c69765126e6c015b182eca, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0008527749999984735>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=NotPrimaryError("The server is in quiesce mode and will shut down, full error: {'topologyVersion': {'processId': ObjectId('66c6975cc22c399da9430ac6'), 'counter': 4}, 'ok': 0.0, 'errmsg': 'The server is in quiesce mode and will shut down', 'code': 91, 'codeName': 'ShutdownInProgress', 'remainingQuiesceTimeMillis': 11394, 'lastCommittedOpTime': Timestamp(0, 0), '$clusterTime': {'clusterTime': Timestamp(1724290912, 2), 'signature': {'hash': b'jr\\x01-\\x1b\\x03\\xec\\xc3\\xcd\\xbb\\x9b\\xdf\\xea>\\xfbU\\xc7h-\\xa7', 'keyId': 7405771237584011288}}, 'operationTime': Timestamp(1724290845, 1)}")>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0006911360000003697>]>
unit-shard-one-0: 01:42:01 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c697683f399d7325a9c592, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0005849489999718571>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=NotPrimaryError("The server is in quiesce mode and will shut down, full error: {'topologyVersion': {'processId': ObjectId('66c69767794eb12b0d22d1d6'), 'counter': 4}, 'ok': 0.0, 'errmsg': 'The server is in quiesce mode and will shut down', 'code': 91, 'codeName': 'ShutdownInProgress', 'remainingQuiesceTimeMillis': 14994, 'lastCommittedOpTime': Timestamp(0, 0), '$clusterTime': {'clusterTime': Timestamp(1724290920, 3), 'signature': {'hash': b'e\\x06\\xb0\\x10\\xd70\\x1c\\x9a\\xc6#\\xeb(\\x8aF\\xdc\\xc7cY\\x99\\x01', 'keyId': 7405771237584011288}}, 'operationTime': Timestamp(1724290845, 1)}")>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0006600989999014928>]>
unit-shard-one-0: 01:43:11 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:43:11 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:43:13 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:43:13 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:43:32 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:43:32 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:43:32 ERROR unit.shard-one/2.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:43:32 ERROR unit.shard-one/2.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:43:34 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:43:35 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:45:27 ERROR unit.shard-two/0.juju-log certificates:6: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:45:27 ERROR unit.shard-two/0.juju-log certificates:6: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:47:07 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:47:07 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:47:09 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:47:09 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:47:11 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:47:11 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:47:11 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:47:11 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 01:47:27 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:47:28 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:49:27 ERROR unit.config-server/1.juju-log certificates:7: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:49:27 ERROR unit.config-server/1.juju-log certificates:7: This operation (delete_relation_data()) can only be performed by the leader unit
unit-self-signed-certificates-0: 01:49:35 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "06f94a18-468a-4868-8631-1539dec83517:r#6#shard-two": settings 06f94a18-468a-4868-8631-1539dec83517:r#6#shard-two not found (not found)
unit-shard-two-0: 01:49:42 ERROR unit.shard-two/0.juju-log Unable to access primary due to: shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c69935bdb0a3f254ed7e2b, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-two-0.shard-two-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-shard-one-0: 01:49:43 ERROR unit.shard-one/0.juju-log Unable to access primary due to: shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c699369e8c5c31db943101, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-config-server-0: 01:49:44 ERROR unit.config-server/0.juju-log config-server:4: Deferring _on_relation_event for shards interface since: error=ServerSelectionTimeoutError("config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c699376099de2f5c56c5fb, topology_type: Unknown, servers: [<ServerDescription ('config-server-0.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('config-server-1.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>")
unit-shard-one-0: 01:49:45 ERROR unit.shard-one/0.juju-log Unable to access primary due to: shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c69938a00612b55c6d543f, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-config-server-0: 01:49:47 ERROR unit.config-server/0.juju-log Deferring _on_relation_event for shards interface since: error=ServerSelectionTimeoutError("config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c6993ab0ead4a8a4d0a345, topology_type: Unknown, servers: [<ServerDescription ('config-server-0.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('config-server-1.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>")
unit-shard-two-0: 01:49:49 ERROR unit.shard-two/0.juju-log Unable to access primary due to: shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c6993c33cdd3c2aab34450, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-two-0.shard-two-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-config-server-0: 01:49:49 ERROR unit.config-server/0.juju-log config-server:3: Deferring _on_relation_event for shards interface since: error=ServerSelectionTimeoutError("config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c6993c57debdbd1dc57d89, topology_type: Unknown, servers: [<ServerDescription ('config-server-0.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('config-server-1.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>")
unit-shard-one-0: 01:50:02 ERROR unit.shard-one/0.juju-log sharding:3: Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c69949e656819ee052aadf, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.001964939000117738>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.002512929999966218>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.004988228000001982>]>
unit-shard-one-1: 01:50:11 ERROR unit.shard-one/1.juju-log shard-one/1.is_unit_in_replica_set PyMongoError=No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c699521f3e5e1035ea841d, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: RSGhost, rtt: 0.0043418760001259216>]>
unit-config-server-1: 01:50:12 ERROR unit.config-server/1.juju-log config-server/1.is_unit_in_replica_set PyMongoError=No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66c6995327dbef6ca106c981, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('config-server-1.config-server-endpoints', 27017) server_type: RSGhost, rtt: 0.002807669000048918>]>
controller-0: 01:51:39 ERROR juju.worker.caasapplicationprovisioner.runner exited "config-server": finding filesystem info for config-server-mongodb-9cea78c8: unable to get persistent volume claim: k8s: persistentvolumeclaims "config-server-mongodb-9cea78c8-config-server-1" not found
unit-shard-one-0: 01:53:28 ERROR unit.shard-one/0.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 01:53:28 ERROR unit.shard-one/2.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-two-0: 01:53:28 ERROR unit.shard-two/0.juju-log certificates:12: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 01:53:28 ERROR unit.config-server/1.juju-log certificates:13: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 01:53:28 ERROR unit.shard-one/2.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:53:28 ERROR unit.config-server/0.juju-log certificates:13: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 01:53:29 ERROR unit.config-server/1.juju-log certificates:13: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-two-0: 01:53:29 ERROR unit.shard-two/0.juju-log certificates:12: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-0: 01:53:29 ERROR unit.shard-one/0.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-1: 01:53:29 ERROR unit.shard-one/1.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-0: 01:53:29 ERROR unit.config-server/0.juju-log certificates:13: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-1: 01:53:30 ERROR unit.shard-one/1.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:54:50 ERROR unit.config-server/0.juju-log config-server:15: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291689, 1), \'signature\': {\'hash\': b\'D4J\\xd3\\x9c\\x86\\xc2\\x10h\\x7fm\\xf2\\xb6\\xbba>\\xa1)\\xacn\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291689, 1)}')
unit-config-server-0: 01:54:50 ERROR unit.config-server/0.juju-log config-server:15: Failed to add shard-one to cluster
unit-config-server-0: 01:54:50 ERROR unit.config-server/0.juju-log config-server:15: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291689, 1), \'signature\': {\'hash\': b\'D4J\\xd3\\x9c\\x86\\xc2\\x10h\\x7fm\\xf2\\xb6\\xbba>\\xa1)\\xacn\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291689, 1)}')
unit-config-server-0: 01:54:51 ERROR unit.config-server/0.juju-log config-server:15: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291691, 2), \'signature\': {\'hash\': b\'\\x81q_\\xa4M\\xb9\\xe0\\xe9\\xc0\\x90\\x96Ug\\t\\xc5\\xd0A+\\x05\\xbb\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291691, 2)}')
unit-config-server-0: 01:54:52 ERROR unit.config-server/0.juju-log config-server:15: Failed to add shard shard-two to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291692, 1), \'signature\': {\'hash\': b\'\\xa1\\x13\\xcd\\xd3\\x15\\xcf\\xae\\xdcP\\x03\\n[x\\x01\\x0c\\xe9\\xd7ig*\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291692, 1)}')
unit-config-server-0: 01:54:52 ERROR unit.config-server/0.juju-log config-server:15: Failed to add shard-two to cluster
unit-config-server-0: 01:54:52 ERROR unit.config-server/0.juju-log config-server:15: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=d689c498-f810-42f1-b7fd-c8c95cf2cbb9,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291692, 1), \'signature\': {\'hash\': b\'\\xa1\\x13\\xcd\\xd3\\x15\\xcf\\xae\\xdcP\\x03\\n[x\\x01\\x0c\\xe9\\xd7ig*\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291692, 1)}')
unit-shard-one-1: 01:55:07 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:55:07 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:55:08 ERROR unit.shard-two/0.juju-log certificates:12: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:55:08 ERROR unit.shard-two/0.juju-log certificates:12: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:55:08 ERROR unit.shard-one/2.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:55:08 ERROR unit.shard-one/2.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:55:09 ERROR unit.config-server/0.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291708, 3), \'signature\': {\'hash\': b\'\\x9c\\x88\\xf0\\xe7\\x9fo\\x98\\xb2\\xc0|\\x8a\\x1b\\xcb\\xea\\xf1\\xca\\xc0^\\xed\\x07\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291708, 3)}')
unit-shard-one-0: 01:55:09 ERROR unit.shard-one/0.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:55:10 ERROR unit.shard-one/0.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:55:10 ERROR unit.config-server/0.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-0: 01:55:10 ERROR unit.config-server/0.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291708, 3), \'signature\': {\'hash\': b\'\\x9c\\x88\\xf0\\xe7\\x9fo\\x98\\xb2\\xc0|\\x8a\\x1b\\xcb\\xea\\xf1\\xca\\xc0^\\xed\\x07\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291708, 3)}')
unit-config-server-0: 01:55:26 ERROR unit.config-server/0.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291725, 1), \'signature\': {\'hash\': b\'F\\xff\\xd9.\\x93J\\x05\\x95 %\\x02\\x08\\xdd+\\x7f\\xe2?\\x9e\\x81\\x91\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291725, 1)}')
unit-config-server-0: 01:55:26 ERROR unit.config-server/0.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-0: 01:55:26 ERROR unit.config-server/0.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1724291725, 1), \'signature\': {\'hash\': b\'F\\xff\\xd9.\\x93J\\x05\\x95 %\\x02\\x08\\xdd+\\x7f\\xe2?\\x9e\\x81\\x91\', \'keyId\': 7405775811724181529}}, \'operationTime\': Timestamp(1724291725, 1)}')
unit-shard-one-1: 01:58:38 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:58:38 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-2: 02:03:27 ERROR unit.shard-one/2.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 02:03:27 ERROR unit.shard-one/2.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-1: 02:03:27 ERROR unit.shard-one/1.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 02:03:28 ERROR unit.shard-one/1.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-0: 02:03:34 ERROR unit.shard-one/0.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 02:03:35 ERROR unit.shard-one/0.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 02:08:05 ERROR unit.config-server/1.juju-log certificates:13: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:08:05 ERROR unit.config-server/1.juju-log certificates:13: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:10:06 ERROR unit.config-server/1.juju-log certificates:17: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-0: 02:10:07 ERROR unit.config-server/0.juju-log certificates:17: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 02:10:07 ERROR unit.config-server/1.juju-log certificates:17: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 02:10:07 ERROR unit.config-server/0.juju-log certificates:17: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 02:10:10 ERROR unit.config-server/1.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:10:10 ERROR unit.config-server/1.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:10:24 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:10:24 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:10:35 ERROR unit.config-server/1.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:10:35 ERROR unit.config-server/1.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-shard-two-0: 02:12:09 ERROR unit.shard-two/0.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.
unit-shard-one-1: 02:12:10 ERROR unit.shard-one/1.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.
unit-shard-one-0: 02:12:12 ERROR unit.shard-one/0.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.
unit-shard-one-2: 02:12:12 ERROR unit.shard-one/2.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.

[32mINFO    [0m pytest_operator.plugin:plugin.py:904 Forgetting main...